src/all/mangaball/src/eu/kanade/tachiyomi/extension/all/mangaball/MangaBall.kt

```diff
@@ -0,0 +1,409 @@
+                "Ongoing" -> SManga.ONGOING
+                "Completed" -> SManga.COMPLETED
+                "Hiatus" -> SManga.ON_HIATUS
+                "Cancelled" -> SManga.CANCELLED
+                else -> SManga.UNKNOWN
+            }
+        }
+    }
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("-")
+        val body = FormBody.Builder()
+            .add("title_id", id)
+            .build()
+
+        return POST("$baseUrl/api/v1/chapter/chapter-listing-by-title-id/", headers, body)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        (response.request.body as FormBody).also {
+            updateViews(it.value(0))
+        }
+
+        val data = response.parseAs<ChapterListResponse>()
+        val numberFormatter = DecimalFormat("#.##")
+
+        return data.chapters.flatMap { chapter ->
+            chapter.translations.mapNotNull { translation ->
+                if (translation.language in siteLang) {
+                    SChapter.create().apply {
+                        url = translation.id
+                        name = buildString {
+                            if (translation.volume > 0) {
+                                append("Vol. ")
+                                append(translation.volume)
+                                append(" ")
+                            }
+                            val number = numberFormatter.format(chapter.number)
+                            if (translation.name.contains(number)) {
+                                append(translation.name.trim())
+                            } else {
+                                append("Ch. ")
+                                append(number)
+                                append(" ")
+                                append(translation.name.trim())
+                            }
+                        }
+                        chapter_number = chapter.number
+                        date_upload = dateFormat.tryParse(translation.date)
+                        scanlator = buildString {
+                            append(translation.group.name)
+                            // id is usually the name of the site the chapter was scraped from
+                            // if not then it is generated id of an active group on the site
+                            if (groupIdRegex.matchEntire(translation.group.id) == null) {
+                                append(" (")
+                                append(translation.group.id)
+                                append(")")
+                            }
+                        }
+                    }
+                } else {
+                    null
+                }
+            }
+        }
+    }
+
+    private val groupIdRegex = Regex("""[a-z0-9]{24}""")
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH)
```

Comment:
This is more of a style suggestion. The locale isn't used in the pattern.
```suggestion
    private val dateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ROOT)
```

Reply:
i thought we should avoid to use .ROOT?

Reply:
I didn't get the memo, where was that discussed?

Setting the locale here doesn't change anything about the pattern, so to me it seems like unnecessary extra info.